BlueDBOnDisk/src/main/java/io/bluedb/util/CachedSingleThreadingPool.java

```diff
@@ -0,0 +1,75 @@
+package io.bluedb.util;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class CachedSingleThreadingPool {
+
+	
+	private ConcurrentHashMap<String, ConcurrentLinkedQueue<FutureTask<Void>>> collectionTaskMap = new ConcurrentHashMap<>();

```

Comment:
Looks like you're only accessing this from within synchronized blocks so we probably don't need to use concurrent classes. I might look a little more into the synchronized stuff and see if we can slim down the critical sections at all. That being said, everything inside of the synchronized blocks is super fast, so its probably fine.